### PR TITLE
Add FFI for CustomEvent constructor

### LINKED ISSUE #12: Missing constructor for CustomEvent
It would seem that https://github.com/purescript-web/purescript-web-events/blob/7f19f24d7b61066bef0813734964f2dc6b085e3d/src/Web/Event/CustomEvent.purs is missing a binding for the `CustomEvent` constructor, effectively preventing its creation.
I may be mistaken, but I don't see how one is supposed to create a `CustomEvent`other than through `new CustomEvent()`.

### LINKED ISSUE #14: Missing detail getter function for CustomEvent
There is no getter function for `detail` property of `CustomEvent` type. Related to #13.

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Fixes #12. Fixes #14 

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [x] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
:ping_pong: @thomashoneyman

**Comment by garyb** [comment]
I think creating an event should probably happen in `Effect`?

Should probably accept an `EventType` rather than a `String` argument.

File: src/Web/Event/CustomEvent.purs [hunk 1]
```diff
@@ -12,3 +13,15 @@ fromEvent = unsafeReadProtoTagged "CustomEvent"
 
 toEvent :: CustomEvent -> Event
 toEvent = U.unsafeCoerce
+
+foreign import new :: String -> CustomEvent
+
+-- | Create a new `CustomEvent`, storing some data in its `detail` field,
+-- | and using defaults for everything else.
+new' :: forall a. String -> Nullable a -> CustomEvent
```
Comment (thomashoneyman) [inline src/Web/Event/CustomEvent.purs:21]: We generally don't expose `Nullable` directly when we can expose `Maybe` instead. Can we expose `Maybe` interfaces for things exported from this module? (Nullable is fine for FFI internals)

File: src/Web/Event/CustomEvent.purs [hunk 2]
```diff
@@ -12,3 +13,15 @@ fromEvent = unsafeReadProtoTagged "CustomEvent"
...
+new' ty det = newOptions ty { detail: det, bubbles: false, cancelable: false, composed: false }
+
+-- | Create a new `CustomEvent` with all of its constructor's options exposed.
+foreign import newOptions :: forall a. String -> { detail :: Nullable a , bubbles :: Boolean, cancelable :: Boolean, composed :: Boolean } -> CustomEvent
```
Comment (thomashoneyman) [inline src/Web/Event/CustomEvent.purs:25]: If we're going to do both an options variant and a non-options variant, then perhaps we should stick to the same naming scheme we've been using recently, ie. `newWithOptions`.

File: src/Web/Event/CustomEvent.purs [hunk 3]
```diff
@@ -12,3 +13,15 @@ fromEvent = unsafeReadProtoTagged "CustomEvent"
...
+
+foreign import detail :: forall a. CustomEvent -> Nullable a
```
Comment (thomashoneyman) [inline src/Web/Event/CustomEvent.purs:27]: Same note here wrt `Nullable`

**Comment by garyb** [comment]
Using `a` for `detail` is suspect too I think. Certainly it's not suitable for the return type of `detail` anyway, it's `unsafeCoerce` at that point.

**Comment by thomashoneyman** [comment]
> Using `a` for `detail` is suspect too I think. Certainly it's not suitable for the return type of `detail` anyway, it's `unsafeCoerce` at that point. As much as I don't like `Foreign`, I think that's probably the only suitable type we have for it.

Yes, I agree.

**Comment by JordanMartinez** [comment]
Done!

I'll also need to update the spago dependency to include `foreign` in the package set.

**Review by thomashoneyman — APPROVED** [review]

**Pull request merged**